plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/Compiler.java

```diff
@@ -30,125 +30,134 @@
-        Loader(ClassLoader parent) {
+        /**
+         * @param parent The parent ClassLoader.
+         */
+        Loader(final ClassLoader parent) {
             super(parent);
         }
 
-        Class<? extends Executable> define(String name, byte[] bytes) {
+        /**
+         * Generates a Class object from the generated byte code.
+         * @param name The name of the class.
+         * @param bytes The generated byte code.
+         * @return A Class object extending {@link Executable}.
+         */
+        Class<? extends Executable> define(final String name, final byte[] bytes) {
             return defineClass(name, bytes, 0, bytes.length, CODESOURCE).asSubclass(Executable.class);
         }
     }
 
-    static Executable compile(Loader loader, final String name, final String source, final Definition custom, CompilerSettings settings) {
-        long start = System.currentTimeMillis();
-
-        final Definition definition = custom == null ? DEFAULT_DEFINITION : new Definition(custom);
-
-        //long end = System.currentTimeMillis() - start;
-        //System.out.println("types: " + end);
-        //start = System.currentTimeMillis();
-
-        //final ParserRuleContext root = createParseTree(source, types);
-        final ANTLRInputStream stream = new ANTLRInputStream(source);
-        final ErrorHandlingLexer lexer = new ErrorHandlingLexer(stream);
-        final PlanAParser parser = new PlanAParser(new CommonTokenStream(lexer));
-        final ParserErrorStrategy strategy = new ParserErrorStrategy();
-
-        lexer.removeErrorListeners();
-        lexer.setTypes(definition.structs.keySet());
-
-        //List<? extends Token> tokens = lexer.getAllTokens();
-
-        //for (final Token token : tokens) {
-        //    System.out.println(token.getType() + " " + token.getText());
-        //}
+    /**
+     * Runs the two-pass compiler to generate a Plan A script.
+     * @param loader The ClassLoader used to define the script.
+     * @param name The name of the script.
+     * @param source The source code for the script.
+     * @param settings The CompilerSettings to be used during the compilation.
+     * @return An {@link Executable} Plan A script.
+     */
+    static Executable compile(final Loader loader, final String name, final String source,
+                              final Definition custom, final CompilerSettings settings) {
+        // Set the definition.
+        final Definition definition = custom != null ? new Definition(custom) : DEFAULT_DEFINITION;
 
-        parser.removeErrorListeners();
-        parser.setErrorHandler(strategy);
-
-        ParserRuleContext root = parser.source();
-
-        //end = System.currentTimeMillis() - start;
-        //System.out.println("tree: " + end);
-
-        final Adapter adapter = new Adapter(definition, source, root, settings);
-
-        start = System.currentTimeMillis();
+        // Generate the ANTLR tree from the given source.
```

Comment:
I don't think these comments are really required, especially because you name the methods good stuff like `createParseTree`.